packages/blockchain-sdk-aptos/src/auth/ed25519/aptos-ed25519-token-validator.ts

```diff
@@ -0,0 +1,44 @@
+import { sign } from 'tweetnacl';
+
+import { APTOS_ED25519_TOKEN_SIGNER_ALG } from './aptos-ed25519-token-signer';
+import { getAptosAccountAddress } from '../../utils/aptos-account-utils';
+import {
+  AuthenticationError,
+  Token,
+  TokenHeader,
+  TokenValidator,
+} from '@dialectlabs/sdk';
+import { HexString } from 'aptos';
+
+export class AptosEd25519TokenValidator extends TokenValidator {
+  canValidate(tokenHeader: TokenHeader): boolean {
+    return tokenHeader.alg === APTOS_ED25519_TOKEN_SIGNER_ALG;
+  }
+
+  override isSignatureValid(token: Token): boolean {
+    const signedPayload = token.base64Header + '.' + token.base64Body;
+    const signingPayload = new TextEncoder().encode(signedPayload);
+    return sign.detached.verify(
+      signingPayload,
+      token.signature,
+      this.getPublicKey(token),
+    );
+  }
+
+  override performExtraValidation(token: Token): boolean {
```

Comment:
Personal preference but I'd probably name this `extraValidationNeeded` because it's a boolean.

Reply:
I think `extraValidationNeeded` also not the best naming, since the method actually doesn't check if validation is needed, it really validates smth

May be you have alternative naming ideas?

Reply:
Ahh ok. In that case i'd just leave it alone.